CONTRIBUTING.md

```diff
@@ -48,6 +48,30 @@ All pull requests need to abide by the following criteria to be accepted:
 - support for localization in code (translations need not be provided)
 - tests for added / changed functionality
 
+## Working with Git
```

Comment:
I feel like this whole contributing.md change could just be its own PR, this PR is already really large as it is